Setup support for dynamic QML plugins

LeonMatthesKDAB opened the pull request:
Supporting dynamic QML plugins means mostly copying the dynamic library and qmldir files to the right locations.
This patch makes cxx_qt_add_qml_module set this up automatically, if the Rust crate is built to a dynamic library.

Commit:
Only link Qt and initializers to static crates

Dynamic libraries are already linked by Cargo and don't require these
additional steps.

Commit:
Allow importing shared QML modules

This will then copy over the appropriate files to the OUTPUT_DIR, so
that the plugin can be dynamically loaded at runtime.

LeonMatthesKDAB commented on cmake/CxxQt.cmake:
```diff
@@ -181,21 +185,65 @@ function(cxx_qt_import_qml_module target)
+    endif()
+    set(QML_PLUGIN_TARGET "${QML_MODULE_OUTPUT_DIR}/${module_name}/${QML_PLUGIN_NAME}")
+
+    # Finally, copy everything
```
We already point qmlls/qmllint towards our internal target dir (where we copy out of), which contains the qml files.
I would argue by default you should use that for development, but not for deployment.

The directory where we copy the shared library and qmldir file is used for deployment.
As you don't really want to ship your qml code by accident it actually makes sense to keep them separate.

ahayzen-kdab replied:
Right, but guess we might need to document how you are supposed to "install" the module for other developers. As eg if you look at the Qt install folder under `qml/` you'll see most folders have a `.qmltypes` and some have `.qml` where they do on the source side. Otherwise when you run qmllint / qmlls they become confused about these installed types.

Commit:
Copy the other way around

First copy the shared lib into the export directory, then copy the
entire qml module back to the output dir.

ahayzen-kdab reviewed (approved):
LGTM :-)
